letsencrypt-auto-source/letsencrypt-auto.template

```diff
@@ -229,23 +229,40 @@ DeterminePythonVersion() {
 {{ bootstrappers/smartos.sh }}
 {{ bootstrappers/mageia_common.sh }}
 
-# Install required OS packages:
-Bootstrap() {
-  if [ "$NO_BOOTSTRAP" = 1 ]; then
-      return
-  elif [ -f /etc/debian_version ]; then
+# Set Bootstrap to the function that installs OS dependencies on this system
+# and BOOTSTRAP_VERSION to the unique identifier for the current version of
+# that function.
+if [ "$NO_BOOTSTRAP" = 1 ]; then
+  Bootstrap() {
+    :
+  }
+  BOOTSTRAP_VERSION="NO_BOOTSTRAP 1"
+elif [ -f /etc/debian_version ]; then
+  Bootstrap() {
     BootstrapMessage "Debian-based OSes"
     BootstrapDebCommon
-  elif [ -f /etc/mageia-release ]; then
-    # Mageia has both /etc/mageia-release and /etc/redhat-release
+  }
+  BOOTSTRAP_VERSION="BootstrapDebCommon $BOOTSTRAP_DEB_COMMON_VERSION"
+elif [ -f /etc/mageia-release ]; then
+  # Mageia has both /etc/mageia-release and /etc/redhat-release
+  Bootstrap() {
     ExperimentalBootstrap "Mageia" BootstrapMageiaCommon
-  elif [ -f /etc/redhat-release ]; then
+  }
+  BOOTSTRAP_VERSION="BootstrapMageiaCommon $BOOTSTRAP_MAGEIA_COMMON_VERSION"
```

Comment:
> Yes, I tend to branch off into Python at my earliest opportunity (as you can see here) to get some decent flow control.

Yeah. Just to state this explicitly, we can't do that here because this code runs before bootstrapping meaning we can't rely on Python being installed.

The only approaches I have really just push things around and result in more code overall. We could do something like set `BOOTSTRAP_FUNC="BootstrapSuchAndSuch"` and then later for every OS run `"Get"$BootstrapSuchAndSuch"Version"` which could be a function that returns/sets the bootstrap version for that system but I'm personally not convinced that's any cleaner.